### PR TITLE
Feature/login route implementation

### PR DESCRIPTION
**Pull request by beviggiano**
This pull request implements authentication and user validation functionalities including JWT token management, cookie handling, and email/password validation.

### CODE REVIEW

File: backend/src/infra/api/middlewares/auth.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+export async function login(req: Request, res: Response, next: NextFunction) {
+    try {
+        const user = await UserModel.findOne({
+            where: {
+                email: req.body.email
+            }
+        });
+
+        if (!user) {
+            throw new PermissionError("Incorrect email and/or password!");
```
Comment (edu-correia) [inline backend/src/infra/api/middlewares/auth.ts:63]: Please use a more specific error message here, something like: "Could not find this user"
And change it from a PermissionError to a LoginError or whatever you think suits it better

**Review by edu-correia — CHANGES_REQUESTED** [review]
:warning: **Important:** Please consider the comments I made in PR #11 in this commit too.
Very clever authentication implementation, well done :clap:

File: backend/src/infra/api/middlewares/auth.ts (hunk 1 above)
Comment (beviggiano) [reply backend/src/infra/api/middlewares/auth.ts:63]: I ll change to LoginError. But, in relation to use 'could not find this user' I think safer for the users to use the error message not specifying what's wrong with de login. Cause it's more difficult to know if the account really exists, in an attack case.

Comment (edu-correia) [reply backend/src/infra/api/middlewares/auth.ts:63]: Ok, makes sense

**Review by edu-correia — APPROVED** [review]
Looks good to me

**Pull request merged**